django/visions/templates/visions/vision-detail.html

```diff
@@ -0,0 +1,144 @@
+                    </div>
+                    <!-- Additional images -->
+                    {% for additionalimage in object.additionalimages.all %}
+                        <img src="{{ additionalimage.image.url }}" alt="additional vision image" class="vision-detail-content-additionalimage">
+                    {% endfor %}
+                {% endif %}
+                <!-- Video -->
+                {% if object.vision_video %}
+                    <div id="vision-detail-content-video" class="video-container">
+                        {% video object.vision_video as vision_video %}
+                            {% video vision_video %}
+                        {% endvideo %}
+                    </div>
+                {% endif %}
+                <!-- Audio -->
+                <div id="vision-detail-content-audio">
+                    {{ object.vision_audio | safe }}
+                </div>
+            </div>
+        </div>
+
+        <!-- Responses -->
+        <div id="response-create-bg" class="form-hide">
+            <div id="response-create">
+                <!-- Form to create and submit an response to the above vision -->
+                <form enctype="multipart/form-data" id="response-create-form" name="response-create-form" method="POST" action="{% url 'visions:response-create' %}">
+                    <h2>Respond to this Vision</h2>
+                    <!-- Security CSRF token -->
+                    {% csrf_token %}
+                    <!-- Form -->
+                    {{ response_create_form.as_p }}
+                    <!-- Submit form button -->
+                    <button id="response-create-submit" class="btn btn-primary" type="submit">Submit</button>
+                </form>
+                <!-- Form script  -->
+                <script>
+                    $(document).ready(function() {
+                        // Sets the value of the vision in the response form to the current vision id
+                        $('#id_vision').val({{ object.id }}).hide();
+                        // Show form
+                        $('.form-show').on('click', function(){
+                            $('#response-create-bg').show();
+                        });
+                        // Hide form
+                        $('.form-hide').on('click', function(){
+                            $(this).hide();
+                        }).children().on('click', function(e) {
+                            e.stopPropagation();
+                        });
+                    });
+                </script>
+            </div>
+        </div>
+        
+        <!-- List of all users' responses -->
+        <div id="response-list">
+            <h2>Responses</h2>
+
+            <div class="form-show create-form-button-link"><i class="fas fa-comments"></i> Respond to this Vision</div>
```

Comment:
Is the space before 'Respond' intentional? I expect this to have an effect on the way the button looks.

Reply:
Yeah, the <i> tag before 'Respond' is an icon (see https://fontawesome.com/v5/search) so it's nice to separate the icon from the text a bit.